extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesConfig.java

```diff
@@ -230,10 +224,41 @@ public boolean isAddBuildTimestamp() {
         return addBuildTimestamp;
     }
 
+    @Override
+    public String getTarget() {
+        return Constants.KUBERNETES;
+    }
+    
+    /**
+     * Environment variables to add to all containers using the old syntax.
+     *
+     * @deprecated Use {@link #env} instead using the new syntax as follows:
+     * <ul>
+     *     <li>{@code quarkus.kubernetes.env-vars.foo.field=fieldName} becomes {@code quarkus.kubernetes.env.fields.foo=fieldName}</li>
+     *     <li>{@code quarkus.kubernetes.env-vars.envvar.value=value} becomes {@code quarkus.kubernetes.env.vars.envvar=value}</li>
```

Comment:
Neatpicking, but `envar` as a a key here can be confusing.
```suggestion
     *     <li>{@code quarkus.kubernetes.env-vars.foo.value=value} becomes {@code quarkus.kubernetes.env.vars.foo=bar}</li>
```